### PR TITLE
39 gyroscope subsystem

### PR DESCRIPTION
**Pull request by noiritgc**

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -74,8 +82,9 @@ public RobotContainer() {
    * edu.wpi.first.wpilibj2.command.button.JoystickButton}.
    */
   private void configureButtonBindings() {
-    JoystickButton lightButton = new JoystickButton(this.leftJoystick, 1);
-    lightButton.whileTrue(this.m_blinkinCommand);
+    JoystickButton lightButton = new JoystickButton(this.leftJoystick, ControllerConstants.kBalanceButton);
+    lightButton.whileTrue(this.m_balancecommand);
```
Comment (JavaBlendPhilippe) [inline src/main/java/frc/robot/RobotContainer.java:90]: Would change name to not be light button unless the primary purpose of this button is the lights. Just reading it looks more like this button is for balancing, so the name lightButton is very misleading
```suggestion
    JoystickButton balanceButton = new JoystickButton(this.leftJoystick, ControllerConstants.kBalanceButton);
    balanceButton.whileTrue(this.m_balancecommand);
```

File: src/main/java/frc/robot/commands/BalanceCommand.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        // the formula that Noirit used, condensed down (even more now)
+        speed = angle * GyroConstants.kBalanceSpeed;
+        m_ledsubsystem.setRed();
+        m_drivesubsystem.tankDrive(speed, speed);
+    }
+    else {
+        m_drivesubsystem.stop();
+        m_ledsubsystem.setGreen();
+    }
+    /*if(angle >= 5 && angle < 10){
```
Comment (JavaBlendPhilippe) [inline src/main/java/frc/robot/commands/BalanceCommand.java:54]: No Explanation for why this commented out code should be saved, Noirit says the stuff above is the working one, remove this it takes a lot of line space. Either comment explaining its use case or remove.

**Review by JavaBlendPhilippe — CHANGES_REQUESTED** [review]
Looking pretty solid, left some comments on stuff that should be changed, fix those real quic

File: src/main/java/frc/robot/commands/BalanceCommand.java [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    else if(angle <=-35 && angle > -40){
+      speed = -0.40;
+      m_drivesubsystem.tankDrive(speed, speed);
+      m_ledsubsystem.setRed();
+    }
+    
+    else{
+        m_drivesubsystem.stop();
+        m_ledsubsystem.setGreen();
+    }*/
```
Comment (dCellar) [inline src/main/java/frc/robot/commands/BalanceCommand.java:126]: Remove this comment
```suggestion
```

**Review by Fyrefyte — APPROVED** [review]

**Pull request merged**